### PR TITLE
Added automatic protocol selection and added AllowLanInterfaces,DisableDNSLeakProtection, DisableIPv6LeakProtection settings  

### PR DESCRIPTION
**Pull request by renishnair-viper**
- Implemented the option to allow or block local network interfaces while connected to a VPN server.
- Added the option to disable DNS leak protection
- In build.cmd file, updated the cake arguments using double-dash(--)
- Included the ability to disable IPV6 leak protection
- Updated SDK library to 2.2.29
- Updated code to preserve vpnhostservice on installation
- Updated OpenVPN.Binaries.Scramble to 2.5.1

### CODE REVIEW

**Review by jmithy — APPROVED** [review]

**Review by rastapopoulos1 — APPROVED** [review]

File: src/App/Extensions/SDKExtensions.cs [hunk 1]
```diff
@@ -177,6 +168,35 @@ private static async Task Connect(ISDK sdk, ILocation SpecificLocation = null, b
...
+            if (Properties.Settings.Default.ConnectionProtocol == NetworkConnectionType.Automatic)
+            {
+                if (locations == null)
+                {
+                    await sdk.Connect(location, configurations, token); 
+                }
+                else 
+                { 
+                    await sdk.Connect(locations, configurations, token); 
+                }
```
Comment (vimalraj-adikesavan) [inline src/App/Extensions/SDKExtensions.cs]: Simplify this using ?? null-coalescing operator

`await sdk.Connect(locations ?? location, configurations, token);`

File: src/App/Extensions/SDKExtensions.cs [hunk 2]
```diff
@@ -177,6 +168,35 @@ private static async Task Connect(ISDK sdk, ILocation SpecificLocation = null, b
...
+                // Else, connect to a VPN server using a specific Network connection type.
+                var connectionType = configurations.First(x => x.ConnectionType == Properties.Settings.Default.ConnectionProtocol);
+                if (locations == null)
+                {
+                    await sdk.Connect(location, connectionType, token);
+                }
+                else
+                {
+                    await sdk.Connect(locations, connectionType, token);
+                }
```
Comment (vimalraj-adikesavan) [inline src/App/Extensions/SDKExtensions.cs]: Simplify this using ?? null-coalescing operator

File: src/App/Extensions/SDKExtensions.cs (hunk 1 above)
Comment (renishnair-viper) [reply src/App/Extensions/SDKExtensions.cs]: Added null-coalescing operator

File: src/App/Extensions/SDKExtensions.cs (hunk 2 above)
Comment (renishnair-viper) [reply src/App/Extensions/SDKExtensions.cs]: Added null-coalescing operator

**Review by vimalraj-adikesavan — APPROVED** [review]

**Review by rastapopoulos1 — APPROVED** [review]

**Review by vimalraj-adikesavan — APPROVED** [review]

**Pull request merged**